internal/sci/aws/server_test.go

```diff
@@ -0,0 +1,147 @@
+package aws
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/stretchr/testify/assert"
+	"github.com/substratusai/substratus/internal/sci"
+)
+
+const charset = "abcdefghijklmnopqrstuvwxyz0123456789"
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func randomString(length int, charset string) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[rand.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+func TestGetObjectMd5(t *testing.T) {
+	envAccountID := os.Getenv("AWS_ACCOUNT_ID")
+	if envAccountID == "" {
+		t.Skip("Skipping TestGetObjectMd5 because AWS_ACCOUNT_ID is not set")
+	}
+	envClusterName := os.Getenv("CLUSTER_NAME")
+	if envClusterName == "" {
+		t.Skip("Skipping TestGetObjectMd5 because CLUSTER_NAME is not set")
+	}
+	sess, err := session.NewSession()
+	assert.NoError(t, err)
+
+	s3Client := s3.New(sess)
+	server := &Server{
+		Clients: Clients{
+			S3Client: s3Client,
+		},
+	}
+
+	bucket := "substratus-test-bucket-" + randomString(8, charset)
+	object := "test-object"
+
+	_, err = server.Clients.S3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: &bucket,
+	})
+	assert.NoError(t, err)
+
+	defer server.Clients.S3Client.DeleteBucket(&s3.DeleteBucketInput{Bucket: &bucket})
+
+	// Upload an object
+	_, err = server.Clients.S3Client.PutObject(&s3.PutObjectInput{
+		Bucket: &bucket,
+		Key:    &object,
+		Body:   strings.NewReader("test-data"),
+	})
+	assert.NoError(t, err)
+
+	resp, err := server.GetObjectMd5(context.TODO(), &sci.GetObjectMd5Request{
+		BucketName: bucket,
+		ObjectName: object,
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, resp)
+	if resp != nil {
+		assert.NotEmpty(t, resp.Md5Checksum)
+	}
```

Comment:
TODO: Send http request to upload object.

Reply:
Wrong test to add it but yes. I added that test and a request.